File: frontend/android/app/src/main/java/com/colebeing/ping/PingAuthPlugin.kt
package com.colebeing.ping

import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

/**
 * Local, unpublished plugin — just a JS-to-native handoff for the device token returned by
 * /api/push/register-fcm. Not distributed via npm; registered directly in MainActivity.
 */
@CapacitorPlugin(name = "PingAuth")
class PingAuthPlugin : Plugin() {
    @PluginMethod
    fun storeDeviceToken(call: PluginCall) {
        val value = call.getString("value")
        if (value.isNullOrEmpty()) {
            call.reject("value is required")
            return
        }
        DeviceTokenStore.store(context, value)
        call.resolve()
    }
}

File: frontend/ios/App/App/PingAuthPlugin.swift
import Capacitor

/// Local, unpublished plugin — just a JS-to-native handoff for the device token returned by
/// /api/push/register-fcm. Not distributed via npm; registered in PingBridgeViewController.
/// Mirrors android/.../PingAuthPlugin.kt exactly.
@objc(PingAuthPlugin)
public class PingAuthPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "PingAuthPlugin"
    public let jsName = "PingAuth"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "storeDeviceToken", returnType: CAPPluginReturnPromise)
    ]

    @objc func storeDeviceToken(_ call: CAPPluginCall) {
        guard let value = call.getString("value"), !value.isEmpty else {
            call.reject("value is required")
            return
        }
        DeviceTokenStore.store(value)
        call.resolve()
    }
}
